src/playlist-loader/playlist-loader.js

```diff
@@ -0,0 +1,337 @@
+import logger from '../util/logger';
+import window from 'global/window';
+
+/**
+ * A base class for PlaylistLoaders that seeks to encapsulate all the
+ * shared functionality from dash and hls.
+ *
+ * @extends videojs.EventTarget
+ */
+class PlaylistLoader extends videojs.EventTarget {
+
+  /**
+   * Create an instance of this class.
+   *
+   * @param {string} uri
+   *        The uri of the manifest.
+   *
+   * @param {Object} options
+   *        Options that can be used.
+   *
+   * @param {Object} options.vhs
+   *        The VHS object, used for it's xhr
+   *
+   * @param {Object} [options.manifest]
+   *        A starting manifest object.
+   *
+   * @param {Object} [options.manifestString]
+   *        The raw manifest string.
+   *
+   * @param {number} [options.lastRequestTime]
+   *        The last request time.
+   *
+   * @param {boolean} [options.withCredentials=false]
+   *        If requests should be sent withCredentials or not.
+   *
+   * @param {boolean} [options.handleManifestRedirects=false]
+   *        If manifest redirects should change the internal uri
+   */
+  constructor(uri, options = {}) {
+    super();
+    this.logger_ = logger(this.constructor.name || 'PlaylistLoader');
+    this.uri_ = uri;
+    this.options_ = options;
+    this.manifest_ = options.manifest || null;
+    this.vhs_ = options.vhs;
+    this.manifestString_ = options.manifestString || null;
+    this.lastRequestTime_ = options.lastRequestTime || null;
+
+    this.mediaRefreshTime_ = null;
+    this.mediaRefreshTimeout_ = null;
+    this.request_ = null;
+    this.started_ = false;
+    this.on('refresh', this.refreshManifest_);
+    this.on('updated', this.setMediaRefreshTimeout_);
+  }
+
+  /**
+   * A getter for the current error object.
+   *
+   * @return {Object|null}
+   *         The current error or null.
+   */
+  error() {
+    return this.error_;
+  }
+
+  /**
+   * A getter for the current request object.
+   *
+   * @return {Object|null}
+   *         The current request or null.
+   */
+  request() {
+    return this.request_;
+  }
+
+  /**
+   * A getter for the uri string.
+   *
+   * @return {string}
+   *         The uri.
+   */
+  uri() {
+    return this.uri_;
+  }
+
+  /**
+   * A getter for the manifest object.
+   *
+   * @return {Object|null}
+   *         The manifest or null.
+   */
+  manifest() {
+    return this.manifest_;
+  }
+
+  /**
+   * Determine if the loader is started or not.
+   *
+   * @return {boolean}
+   *         True if stared, false otherwise.
+   */
+  started() {
+    return this.started_;
+  }
+
+  /**
+   * The last time a request happened.
+   *
+   * @return {number|null}
+   *         The last request time or null.
+   */
+  lastRequestTime() {
+    return this.lastRequestTime_;
+  }
+
+  /**
+   * A function that is called to when the manifest should be
+   * re-requested and parsed.
+   *
+   * @listens {PlaylistLoader#updated}
+   * @private
+   */
+  refreshManifest_() {
```

Comment:
The main function of most playlist loaders, called when started or when updated is triggered on itself.